storage: TestGossipFirstRange times out on shutdown

`make stress PKG=./storage TESTS=GossipFirstRange TESTTIMEOUT=30s`

My first attempt at explanation here would be that some process needs the first range to finish an outstanding task, but the first range isn't being gossipped since we're already quiescing.

```
I160802 05:13:40.336502 stopper.go:408  quiesceing; tasks left:
1      gossip/infostore.go:276
E160802 05:13:40.336534 storage/replica_command.go:2973  unable to gossip first range: node unavailable; try another peer
I160802 05:13:40.337104 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:50173->127.0.0.1:50220: use of closed network connection
I160802 05:13:40.338537 /Users/tschottdorf/go/src/google.golang.org/grpc/clientconn.go:444  Failed to dial 127.0.0.1:50173: context canceled; please retry.
I160802 05:13:40.338581 /Users/tschottdorf/go/src/google.golang.org/grpc/clientconn.go:444  Failed to dial 127.0.0.1:50218: grpc: the connection is closing; please retry.
I160802 05:13:40.340495 /Users/tschottdorf/go/src/google.golang.org/grpc/clientconn.go:444  Failed to dial 127.0.0.1:50173: grpc: the connection is closing; please retry.
panic: test timed out after 30s

goroutine 586 [running]:
panic(0x522d820, 0xc8210a1150)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
testing.startAlarm.func1()
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:725 +0x14b
created by time.goFunc
	/usr/local/Cellar/go/1.6.2/libexec/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive]:
testing.RunTests(0x5acbe10, 0x63a41a0, 0x12a, 0x12a, 0xc820249701)
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc8203f5ef8, 0xc82029fc98)
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:515 +0x81
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc8203f5ef8)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:57 +0x260
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:658 +0x114

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/asm_amd64.s:1998 +0x1

goroutine 19 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x63b0880)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1015 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:598 +0x8a

goroutine 73 [semacquire]:
sync.runtime_Semacquire(0xc8205f7c4c)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/sema.go:47 +0x26
sync.(*WaitGroup).Wait(0xc8205f7c40)
	/usr/local/Cellar/go/1.6.2/libexec/src/sync/waitgroup.go:127 +0xb4
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Stop(0xc8205f7c20)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:354 +0x20b
github.com/cockroachdb/cockroach/testutils/testcluster.(*TestCluster).stopServers(0xc820250060)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/testutils/testcluster/testcluster.go:84 +0x109
github.com/cockroachdb/cockroach/testutils/testcluster.(*TestCluster).(github.com/cockroachdb/cockroach/testutils/testcluster.stopServers)-fm()
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/testutils/testcluster/testcluster.go:154 +0x20
github.com/cockroachdb/cockroach/util/stop.CloserFn.Close(0xc8203d7dd0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:86 +0x19
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Stop(0xc82024e090)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:358 +0x2e5
github.com/cockroachdb/cockroach/storage_test.TestGossipFirstRange(0xc82024e000)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/gossip_test.go:122 +0x9ea
testing.tRunner(0xc82024e000, 0x63a5cd0)
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:582 +0x892
```

@tamird Another failure that looks related: https://circleci.com/gh/petermattis/cockroach/1260

In the logs I see:

```
I160803 00:54:42.533666 stopper.go:408  quiesceing; tasks left:
1      ts/db.go:94
I160803 00:54:42.533932 /go/src/google.golang.org/grpc/clientconn.go:444  Failed to dial 127.0.0.1:58366: grpc: the connection is closing; please retry.
I160803 00:54:42.534081 internal/client/db.go:489  failed batch: node unavailable; try another peer
W160803 00:54:42.534166 ts/db.go:101  error writing time series data: node unavailable; try another peer
```

Do we need that `GracefulStop` call?

--- a/gossip/simulation/network.go
+++ b/gossip/simulation/network.go
@@ -104,7 +104,7 @@ func (n *Network) CreateNode() (*Node, error) {
 	}
 	n.Stopper.RunWorker(func() {
 		<-n.Stopper.ShouldQuiesce()
-		server.GracefulStop()
+		netutil.FatalIfUnexpected(ln.Close())
 		<-n.Stopper.ShouldStop()
 		server.Stop()
 	})

--- a/server/server.go
+++ b/server/server.go
@@ -360,7 +360,7 @@ func (s *Server) Start() error {
 
 	s.stopper.RunWorker(func() {
 		<-s.stopper.ShouldQuiesce()
-		s.grpc.GracefulStop()
+		netutil.FatalIfUnexpected(anyL.Close())
 		<-s.stopper.ShouldStop()
 		s.grpc.Stop()
 	})

--- a/util/netutil/net.go
+++ b/util/netutil/net.go
@@ -49,7 +49,7 @@ func ListenAndServeGRPC(
 
 	stopper.RunWorker(func() {
 		<-stopper.ShouldQuiesce()
-		server.GracefulStop()
+		FatalIfUnexpected(ln.Close())
 		<-stopper.ShouldStop()
 		server.Stop()
 	})

